frontend/packages/console-plugin-sdk/src/typings/post-form-submit.ts

```diff
@@ -0,0 +1,19 @@
+import { Extension } from './base';
+
+namespace ExtensionProperties {
+  export interface PostFormSubmissionAction {
+    /** action type */
+    type: string;
+    /** callback for the related action */
+    callback: (...args) => void;
```

Comment:
This should probably be typed as an array of resources since the caller must know what to provide the actions.

Reply:
Fixed, added proper type for extension and hook.